src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,121 @@
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+
+import cn from 'classnames';
+import { SearchParams, getSearchWith } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[];
+  onFilter: (fp: Person[]) => void;
+};
+
+const centuriesList = ['16', '17', '18', '19', '20'];
+
+const getFilteredBySex = (people: Person[], sex: string) => {
+  return sex ? people.filter(person => person.sex === sex) : people;
+};
+
+const getFilteredByQuery = (people: Person[], query: string) => {
+  const prepearedQuery = query.trim().toLowerCase();
+
+  if (!query) {
+    return people;
+  }
+
+  return people.filter(person =>
+    person.name.toLocaleLowerCase().includes(prepearedQuery),
+  );
+};
+
+const getFilteredByCentury = (people: Person[], centuries: string[]) => {
+  if (!centuries.length) {
+    return people;
+  }
+
+  return people.filter(person =>
+    centuries.some(century => Math.ceil(person.born / 100) === +century),
+  );
+};
+
+export const PeopleFilters: React.FC<Props> = ({ people, onFilter }) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = useMemo(
+    () => searchParams.getAll('centuries') || [],
+    [searchParams],
+  );
+
+  const filteredBySex = useMemo(
+    () => getFilteredBySex(people, sex),
+    [people, sex],
+  );
+  const filteredByQuery = useMemo(
+    () => getFilteredByQuery(filteredBySex, query),
+    [filteredBySex, query],
+  );
+  const filteredByCentury = useMemo(
+    () => getFilteredByCentury(filteredByQuery, centuries),
+    [filteredByQuery, centuries],
+  );
+
+  function setSearchWith(params: SearchParams) {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+  }
+
+  function handleQueryChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setSearchWith({ query: event.target.value || null });
+  }
+
+  function toggleCenturis(ch: string) {
+    const newCenturies = centuries.includes(ch)
+      ? centuries.filter(century => century !== ch)
+      : [...centuries, ch];
+
+    setSearchWith({ centuries: newCenturies });
+  }
+
+  function clearCenturies() {
+    setSearchWith({ centuries: null });
+  }
+
+  useEffect(() => {
+    onFilter(filteredByCentury);
+  }, [filteredByCentury, onFilter]);
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
+        <SearchLink
+          className={cn({
+            'is-active': !sex,
+          })}
+          params={{}}
+        >
           All
-        </a>
-        <a className="" href="#/people?sex=m">
+        </SearchLink>
+        <SearchLink
+          className={cn({
+            'is-active': sex === 'm',
```

Comment:
```suggestion
            'is-active': sex === MAN,
```
Consider creating variables instead of magic strings